cli_tools/common/distro/distro.go

```diff
@@ -52,8 +75,21 @@ type Release interface {
 // a Release if the arguments are syntactically correct, and represent a
 // release that we *may* support. The caller is responsible for verifying
 // whether a translator is available elsewhere in the system.
-func FromComponents(distro string, major string, minor string) (r Release, e error) {
+func FromComponents(distro string, major string, minor string, architecture string) (r Release, e error) {
+	if distro == "" {
+		return nil, errors.New("distro name required")
+	}
 	distro = strings.ToLower(distro)
+	architecture = strings.ToLower(architecture)
+
+	if architecture != "" && !architectures[architecture] {
+		return nil, fmt.Errorf("Architecture `%s` is not supported for import", architecture)
+	}
+
+	if distro == windows {
+		return newWindowsRelease(major, minor, architecture)
+	}
```

Comment:
It would be more readable if windows was handled in the main switch below. Is the break in logic because the rest of the code is Linux specific? 

If it's not possible to have windows in the switch, consider extracting the rest of this func in e.g. newLinuxRelease()

Reply:
Extracted out a method to handle Linux.

(The Linux code shares version validation logic)